app/models/concerns/foreman_remote_execution/host_extensions.rb

```diff
@@ -3,17 +3,45 @@ module HostExtensions
     extend ActiveSupport::Concern
 
     included do
+      alias_method_chain :build_required_interfaces, :remote_execution
+      alias_method_chain :reload, :remote_execution
+      alias_method_chain :becomes, :remote_execution
+
       has_many :targeting_hosts, :dependent => :destroy, :foreign_key => 'host_id'
     end
 
-    # create or overwrite instance methods...
-    def instance_method_name
+    def execution_interface
+      get_interface_by_flag(:execution)
+    end
+
+    def remote_execution_proxies(provider)
+      proxies = {}
+      proxies[:subnet]   = execution_interface.subnet.remote_execution_proxies.with_features(provider) if execution_interface && execution_interface.subnet
+      proxies[:fallback] = smart_proxies.with_features(provider) if Setting[:remote_execution_fallback_proxy]
```

Comment:
I'm not entirely sure what you mean by the host strategy.
Right now we have 3 ways to determine the proxy (subnet, fallback, global) and want to add topology (and/or host?) one?
Too many options might be confusing for some users but I think 5 options might be the middle ground between forcing the user to do things "our" way and being overly complicated.